go/internal/feast/registry.go

```diff
@@ -28,7 +29,7 @@ type Registry struct {
 	cachedFeatureViews         map[string]map[string]*core.FeatureView
 	cachedOnDemandFeatureViews map[string]map[string]*core.OnDemandFeatureView
 	cachedRequestFeatureViews  map[string]map[string]*core.RequestFeatureView
-
+	cachedRegistry             *core.Registry
```

Comment:
I thought the plan was to only keep the cached registry proto, just as in Python - what's the reason for keeping cachedEntities and other maps?

Reply:
It turns out we use the map in getEntity() and each of the other get Functionality. I think it makes sense now to generate the map before hand. 
Specifically, this makes it much faster to retrieve entities individually. Before we would compare every entity in the proto for either the project or the name but now we can just use the map instead.

Reply:
I think this design is probably better and removes latency from registry metadata retrieval and offloads the work of loading the featureviews to the actual cache refresh. 

Specifically, each time GetEntity() or listEntities() is called, it loops through every entity in the registry and compares the project name and entity name to retrieve the specified entities. I think it makes more sense to just keep it as is as a map so then retrieving them is just O(n) in the number of entities/project that matches the name.
What do you think @achals

Reply:
I think exploding out into individual maps is fine in that case. It probably makes access easier (but we should make sure that updates to the individual fields are atomic).

Reply:
Yup, I locked the load functionality to make sure it's atomic.